### DISCUSSION TITLE
Groups error when changing to a doc with non-existing field group

### DISCUSSION THREAD

**Discussion by legowhales** [sanity-io/sanity#3794] [Sanity Studio Support] [answered]
Hey,

I am facing an issue with the latest version (dev-preview.22). [edit] Actually the problem happens with older version as well.

When I go to an another document that has different group fields. Changing to a document that does not have the same group field selected will throw an error.

You can see this in action here: https://www.loom.com/share/b2c51ced26c14944892cb19c8bb672e2

[edit] I didn't know we could embed videos and images here. I added the issue in action here:

https://user-images.githubusercontent.com/19142403/198540801-ce5b0815-f7dc-4db7-b006-0679fafe942c.mov

![Capture d’écran 2022-10-28 à 10 22 05](https://user-images.githubusercontent.com/19142403/198540575-d5f50ac1-82fc-48ff-b887-116d96db772f.png)

**Comment by legowhales** [comment]
I actually found a way to fix this, but I do not know how to contribute to v3.

The fix is in this file: ```/sanity/src/core/form/store/formState.ts```

What is the right way to contribute to v3?

```typescript
// ...
function prepareObjectInputState<T>(
  props: RawState<ObjectSchemaType, T>,
  enableHiddenCheck = true
): ObjectFormNode | null {
  // ...
  const defaultGroupName = (schemaTypeGroupConfig.find((g) => g.default) || ALL_FIELDS_GROUP)?.name
  // add this:
  const schemaTypeGroups = [ALL_FIELDS_GROUP, ...schemaTypeGroupConfig];
  const fieldGroupStateName = schemaTypeGroups.find((g) => g.name == props.fieldGroupState?.value)?.name;
  // then:
  const groups = schemaTypeGroups.flatMap((group): FormFieldGroup[] => {
    // ...
    // change: const isSelected = group.name === (props.fieldGroupState?.value || defaultGroupName)
    // to:
    const isSelected = group.name === (fieldGroupStateName || defaultGroupName)
    // ...
  })
}
```

This code keeps the group selection active between doc change. And if the group field does not exist, it selects the default group or All fields.

**Comment by bjoerge** [comment, marked as answer]
This should be addressed by https://github.com/sanity-io/sanity/pull/4050 which is scheduled for release in the next couple of days. Thank you for an excellent description of the issue!

**Reply by bjoerge** [reply]
Fix released in [v3.2.0](https://github.com/sanity-io/sanity/releases/tag/v3.2.0)

**Reply by legowhales** [reply]
Awesome, thank you so much!
